### PR TITLE
Fix magic link issues.

### PR DESCRIPTION
**Pull request by diegoreymendez**
WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/15557
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/322

Please refer to the WPiOS PR for details and testing steps.

### CODE REVIEW

File: WordPressAuthenticator/Authenticator/WordPressAuthenticator.swift [hunk 1]
```diff
@@ -309,68 +305,91 @@ import AuthenticationServices
...
 
-
     /// Present a signin view controller to handle an authentication link.
     ///
     /// - Parameters:
     ///     - url: The authentication URL
-    ///     - rootViewController: The view controller to act as the presenter for the signin view controller.
-    ///                           By convention this is the app's root vc.
+    ///     - rootViewController: The view controller to act as the presenter for the signin view controller.  By convention this is the app's root vc.
+    ///     - automatedTesting: for calling this method for automated testing.  It won't sync the account or load any other VCs.
```
Comment (Gio2018) [inline WordPressAuthenticator/Authenticator/WordPressAuthenticator.swift:313]: I wonder if there's another way to call this method in unit tests without using a flag? Perhaps we can inject a `Storyboard` instance, which could be nil in case of testing? Not sure if it's possible in this case, but the flag might be harder to maintain and could lead to bugs that are not immediately visible.

File: WordPressAuthenticator/Credentials/WordPressComCredentials.swift [hunk 2]
```diff
@@ -19,17 +19,19 @@ public struct WordPressComCredentials: Equatable {
     /// The site address used during login
     ///
     public var siteURL: String
-
-    /// Designated initializer
+    
+    private let wpComURL = "https://wordpress.com"
```
Comment (Gio2018) [inline WordPressAuthenticator/Credentials/WordPressComCredentials.swift:23]: "https://wordpress.com" is used in multiple locations; maybe not in this PR, but I'm wondering if it makes sense to consolidate the constant in just one place?

**Review by Gio2018 — COMMENTED** [review]
Hey @diegoreymendez , this looks generally good, just had a couple of comments.

File: WordPressAuthenticator/Credentials/WordPressComCredentials.swift (hunk 2 above)
Comment (diegoreymendez) [reply WordPressAuthenticator/Credentials/WordPressComCredentials.swift:23]: While I agree with the observation, it was my feeling that trying to address this as part of these PRs is a bit beyond scope.  I think consolidating these constants requires some analysis of its own that would risk causing scope creep in these PRs.

File: WordPressAuthenticator/Authenticator/WordPressAuthenticator.swift (hunk 1 above)
Comment (diegoreymendez) [reply WordPressAuthenticator/Authenticator/WordPressAuthenticator.swift:313]: I agree this isn't ideal, but it was the best solution in terms of simplicity and scope-tightness that I could find.

In my modest opinion the reason I'm having to do this is because the code isn't well architected.  I could try to inject the code to "continue" but the problem is that this method exists at the boundary between WPiOS and WPAuth... meaning that the logic for spawning the VCs is defined now within WPAuth, and if I made that code injectable, it would need to be moved to WPiOS.

I don't know if that makes sense, but basically injecting logic would mean moving that logic to all host Apps (WCiOS, WPiOS, Woo, etc).

As in the reply below though - I don't particularly disagree... the reason I chose the boolean instead is to cause the smallest possible impact as part of a bugfix (since these PRs are not really enhancements).

Comment (Gio2018) [reply WordPressAuthenticator/Authenticator/WordPressAuthenticator.swift:313]: I see. If the price is to have an impact on all the apps, then we should probably leave as it is for now.

File: WordPressAuthenticator/Credentials/WordPressComCredentials.swift (hunk 2 above)
Comment (Gio2018) [reply WordPressAuthenticator/Credentials/WordPressComCredentials.swift:23]: Totally. I noticed this here but it needs not to be addressed in this PR.

**Review by Gio2018 — APPROVED** [review]
:shipit:

**Comment by diegoreymendez** [comment]
Thanks Gio!
